src/main/scala/fs2/interop/cats/cats.scala

```diff
@@ -24,6 +27,15 @@ package object cats extends Instances {
     def runFoldMap[B](f: A => B)(implicit M: Monoid[B]): Free[F, B] =
       self.runFold(M.empty)((b, a) => M.combine(b, f(a)))
 
+    def runGroupByFoldMap[K, B: Monoid](f: A => K)(g: A => B): Free[F, Map[K, B]] =
+      runFoldMap(a => Map(f(a) -> g(a)))
+
+    def runGroupByFoldMonoid[K](f: A => K)(implicit M: Monoid[A]): Free[F, Map[K, A]] =
+      runFoldMap(a => Map(f(a) -> a))
+
+    def runGroupBy[X[_], K](f: A => K)(implicit A: Applicative[X], M: Monoid[X[A]]): Free[F, Map[K, X[A]]] =
```

Comment:
Yes, I have had the same issue. I explicitly used `List` before the PR, as you can see in functional-streams-for-scala/fs2#584.

I wanted to provide users the opportunity to choose which collection/container they are using for the grouped data. However, with this approach, users will need to specify the type in many/most(/all?) cases - I honestly expected the following would infer correctly from the expected type `Map[Int, List[Int]]`:

```
sbt @ fs2-cats $ console
[info] Starting scala interpreter...
[info] 
import fs2._
import fs2.util.Task
import fs2.interop.cats._
import cats._
import cats.implicits._
Welcome to Scala 2.11.8 (OpenJDK 64-Bit Server VM, Java 1.8.0_92).
Type in expressions for evaluation. Or try :help.

scala> val oddeven: Map[Int, List[Int]] = Stream.range(0, 10).covary[Task].runGroupBy(a => a % 2).run.unsafeRun
<console>:24: error: ambiguous implicit values:
 both value Id in package cats of type => cats.Bimonad[cats.Id] with cats.Traverse[cats.Id]
 and value function0Instance in trait Function0Instances of type => cats.Bimonad[Function0]
 match expected type cats.Applicative[X]
       val oddeven: Map[Int, List[Int]] = Stream.range(0, 10).covary[Task].runGroupBy(a => a % 2).run.unsafeRun
                                                                                     ^
```

I don't know what the best choice would be, considering the trade-off of conciseness versus flexibility.

I guess, I as a user would rather want to specify my type explicitly instead of living with a default one, especially considering, that the normal use case would be to specify a collection like `List` or `Vector` and that this collections inherent performance characteristics might be relevant to the post-processing steps.